diff --git a/twrp.cpp b/twrp.cpp
index 3126faf6..ea65486c 100644
--- a/twrp.cpp
+++ b/twrp.cpp
@@ -424,6 +424,9 @@ int main(int argc, char **argv) {
 	if (startup.Get_Fastboot_Mode()) {
 		LOGINFO("starting fastboot\n");
 		gui_msg(Msg("fastboot_console_msg=Entered Fastboot mode..."));
+		// Check for and run startup script if script exists
+		TWFunc::check_and_run_script("/system/bin/runatboot.sh", "boot");
+		TWFunc::check_and_run_script("/system/bin/postrecoveryboot.sh", "boot");
 		if (gui_startPage("fastboot", 1, 1) != 0) {
 			LOGERR("Failed to start fastbootd page.\n");
 		}
